packages/foundation/src/utils.ts

```diff
@@ -46,3 +59,101 @@ export function calculateFees(
     projectFee: pricingInfo.projectFee * quantity,
   }
 }
+
+export async function getSaleTermsId(
+  mint: MintActionParams,
+  multiTokenAddress: Address,
+) {
+  const { chainId, contractAddress, tokenId } = mint
+  if (tokenId == null) return null
+
+  try {
+    const client = createPublicClient({
+      chain: chainIdToViemChain(chainId),
+      transport: http(),
+    }) as PublicClient
+
+    const result = (await client.readContract({
+      address: multiTokenAddress,
+      abi: [GET_SALE_TERMS_1155],
+      functionName: 'getSaleTermsForToken',
+      args: [contractAddress, tokenId],
+    })) as bigint
+
+    return result
+  } catch {
+    return null
+  }
+}
+
+export async function getFixedPriceSaleTerms(
+  client: PublicClient,
+  salesTermId: bigint,
+  multiTokenAddress: Address,
+): Promise<SaleTerms> {
+  return (await client.readContract({
+    address: multiTokenAddress,
+    abi: [GET_FIXED_PRICE_1155],
+    functionName: 'getFixedPriceSale',
+    args: [salesTermId, REFERRAL_ADDRESS],
+  })) as SaleTerms
+}
+export async function getContractType(
+  client: PublicClient,
+  contractAddress: Address,
+): Promise<'1155' | '721'> {
+  const abi = SUPPORTS_INTERFACE_ABI
+  const interfaceIds = {
+    '1155': '0xd9b67a26', // ERC-1155
+    '721': '0x80ac58cd', // ERC-721
+  }
+
+  for (const [type, interfaceId] of Object.entries(interfaceIds)) {
+    try {
+      const supportsInterface = (await client.readContract({
+        address: contractAddress,
+        abi,
+        functionName: 'supportsInterface',
+        args: [interfaceId],
+      })) as boolean
+
+      if (supportsInterface) {
+        return type as '1155' | '721'
+      }
+      // eslint-disable-next-line no-empty
+    } catch {}
+  }
+  throw new Error('Invalid contract type')
+}
+
+export function formatAmount(amount: FilterOperator | undefined) {
+  if (amount === undefined) {
+    return undefined
+  }
+  if (amount && ['string', 'number', 'bigint'].includes(typeof amount)) {
+    return { $gte: amount }
+  }
+
+  return amount
+}
+
+export function getMintAmount(amount: FilterOperator | undefined) {
```

Comment:
This might actually be worth putting in utils - seems useful for most simulations